src/CharacterStates/CharacterState.h

```diff
@@ -23,15 +26,20 @@ class CharacterState {
 
     virtual void changeState(const Uint8* currentKeyStates, Player* player);
 
-    int getCurrentFrame();
     int getFramesAmount();
 
     std::string getStateType();
 
 protected:
+    static const int NORMAL_FRAME = 0;
```

Comment:
If every state will use a different kind of frame, wouldn't it be better if we simple make FRAME_AMOUNTS and each of the following states replace that value?. Ex: jumping state will use the frame amount 4 and only will have to override the value of FRAME_AMOUNTS instead of using the rest of the frames amounts